### PR TITLE
Update tls guides for cert-manager 1.4.0 and coder 1.20

### PR DESCRIPTION
**Pull request by mterhar**
Fixed a bunch of little issues related to newer cert-manager, k8s, coder, etc.
Also the secret for DNS01 challenges needs to be in cert-manager namespace, not coder.

We may want to change the whole directory name to "TLS" since "SSL" is a dead protocol.

### CODE REVIEW

**Comment by mterhar** [comment]
@bpmct and @ericpaulsen  can you sanity check these updates before I request polishing?

File: guides/ssl-certificates/cloudDNS.md
Comment (jawnsy) [inline guides/ssl-certificates/cloudDNS.md:24]: suggest we link to our internal Kubernetes page instead of directly to the upstream docs, as the versions Coder supports may differ from the versions Kubernetes upstream supports

**Review by jawnsy — COMMENTED** [review]
Thanks for this! Can you also go through and make sure we're not using any v1alpha API versions too?

File: guides/ssl-certificates/cloudDNS.md
Comment (mterhar) [reply guides/ssl-certificates/cloudDNS.md:24]: These are guides so I'd like folks to start with a recommendation that makes the most sense. Even this should probably just say "latest" probably since it's not a document that expresses the limits but tries to get people started from the best point.

Comment (jawnsy) [reply guides/ssl-certificates/cloudDNS.md:24]: Well, the latest version of Kubernetes may not be a version that Coder (both the company & the product) supports. We currently follow the same support policy so it does't make a difference, but in the future it might. I think in general we want to keep people in our docs.

To clarify, I'm suggesting that we link to: https://github.com/cdr/docs/blob/main/setup/kubernetes/index.md rather than the Kubernetes documentation, or omit the new link completely here.

Of course, the content right now is not complete, and will be better once I finish/merge #232

**Comment by mterhar** [comment]
> Thanks for this! Can you also go through and make sure we're not using any v1alpha API versions too?

Good call. I updated those and made sure the examples matched the `cert-manager.io/v1` spec.

File: guides/ssl-certificates/cloudDNS.md
Comment (mterhar) [reply guides/ssl-certificates/cloudDNS.md:24]: That page does make more sense than the current one. I'll leave it to @khorne3 and others which link makes more sense from a holistic docs management perspective.

File: guides/ssl-certificates/cloudflare.md
Comment (bpmct) [reply guides/ssl-certificates/cloudflare.md:90]: I was positive the secret worked well when in `coder`, but wanted to sanity check that it would break if changed to `cert-manager`. It does. Recorded a quick video: https://youtu.be/i1fSP32MmPU

basically:
- if person is using ClusterIssuer, secret needs to be in `cert-manager` namespace
- if person is using Issuer, secret needs to be in sane namespace as Issuer

Comment (bpmct) [reply guides/ssl-certificates/cloudflare.md:90]: I also personally like recommending an Issuer over a ClusterIssuer, so I just suggest reverting this change. However, it seems like all the other guides are `ClusterIssuer` first. If you do that, then we should reverse the lower section comparing the two.

**Review by bpmct — CHANGES_REQUESTED** [review]
Caught an error in the CloudFlare docs. Other than that, LGTM.

File: guides/ssl-certificates/cloudflare.md
Comment (khorne3) [reply guides/ssl-certificates/cloudflare.md:90]: @mterhar Do you have any preferences on recommending Issuer over ClusterIssuer? I'd be happy to make the edits but I'm not sure which route works best for users

Comment (mterhar) [reply guides/ssl-certificates/cloudflare.md:90]: Looks like Ben has discovered a nuance that I was missing.  It's tough to know which will be the least friction for users.  It may make sense to have all of our specs reference "issuer" and set the namespaces for issuer and secrets to "coder" for internal consistency and then reference the clusterissuer secret namespace in a separate `TLS troubleshooting` section.

File: guides/ssl-certificates/azureDNS.md
Comment (jawnsy) [inline guides/ssl-certificates/azureDNS.md:34]: FYI: now that https://github.com/cdr/docs/pull/232 was merged, you can link to that page instead https://coder.com/docs/coder/v1.20/setup/kubernetes#supported-kubernetes-versions

File: guides/ssl-certificates/cloudflare.md
Comment (jawnsy) [reply guides/ssl-certificates/cloudflare.md:90]: @deansheather is also quite well-versed in cert-manager shenanigans, maybe worth getting a second opinion from him

**Review by deansheather — COMMENTED** [review]
I think we should just stick with ClusterIssuer for now, we can revisit and change later.

While we're on the topic of cert-manager though, we shouldn't be using the ingress annotation anymore and should instead be creating the Certificate object manually because 1.21.0 does not ship with an Ingress by default. I think this will still work for now, but the included Ingress is deprecated so it won't work for much longer.

File: guides/ssl-certificates/cloudflare.md
Comment (bpmct) [reply guides/ssl-certificates/cloudflare.md:90]:
```suggestion
  namespace: coder # Your coder deployment namespace
```

I am suggesting this change so we can just merge this doc.

**Review by bpmct — APPROVED** [review]
good for 1.20.0

**Review by khorne3 — APPROVED** [review]

**Pull request merged**
